packages/react-components/src/box/src/Box.tsx

```diff
@@ -0,0 +1,48 @@
+import { ElementType, ForwardedRef, ReactElement, ReactNode, forwardRef } from "react";
+import { InnerPropsToProps, omitProps } from "../../shared";
+
+export interface InnerBoxProps {
+    /**
+    * An HTML element type or a custom React element type to render as.
+    */
+    as?: ElementType;
+    /**
+     * [Slot](?path=/docs/getting-started-slots--page) to render into.
+     */
+    slot?: string;
+    /**
+     * React children
+    */
+    children?: ReactNode;
+    /**
+    * @ignore
+    */
+    forwardedRef: ForwardedRef<any>
+}
+
+export type BoxProps = InnerPropsToProps<InnerBoxProps>;
+
+export function InnerBox(props: InnerBoxProps): ReactElement {
+    const {
+        as: Wrapper = "div",
+        children,
+        forwardedRef,
+        ...rest
+    } = omitProps(props, ["slot"]);
+
```

Comment:
this code looks weird. We accept slot as a property, however we never use it. 

Why do we accept it as a props then? Should i remove it from the interface?

Reply:
This `slot` prop is basically the only reason the Box component exist.

We remove it to ensure it's not rendered on the DOM element but it's pickedup by parent Component who support dynamic slots with (useSlots).

Reply:
I'll remove it from the interface, and tweak the omitProps method the not only expect keyof props, but also any string.